contrib/coredns/policy/policy.go

```diff
@@ -245,7 +250,7 @@ func (p *PolicyPlugin) handlePermit(ctx context.Context, w dns.ResponseWriter,
 		if debugQuery {
 			return p.retDebugInfo(r, w, respDomain, nil, "failed")
 		}
-		return status, err
+		return p.retRcode(w, r, dns.RcodeServerFailure, err)
```

Comment:
Why did you change status to dns.RcodeServerFailure?

Reply:
If err != nil status is RcodeServerFailure, so the logic is not broken.

Reply:
status is used later in the code